src/Bicep.Core/Analyzers/Linter/Rules/BCPL1020.cs

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Analyzers.Interfaces;
+using Bicep.Core.Configuration;
+using Bicep.Core.Parsing;
+using Bicep.Core.Semantics;
+using Bicep.Core.Syntax;
+using Microsoft.Extensions.Configuration;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Bicep.Core.Analyzers.Linter.Rules
+{
+    public sealed class BCPL1020 : LinterRule
+    {
+        private readonly HashSet<string> DisallowedHosts;
+
+        public BCPL1020() : base(
+            code: "BCPL1020",
+            ruleName: "Environment() URL hardcoded",
+            description: "Environment() URLs should not be hardcoded",
+            diagnosticLevel: Diagnostics.DiagnosticLevel.Error,
+            docUri: "https://bicep/linter/rules/BCPL1020")// TODO: setup up doc pages
+        {
+            // create a hashset lookup for hosts
+            this.DisallowedHosts = GetConfiguration(nameof(this.DisallowedHosts), Array.Empty<string>())
+                                    .Select( s => s.ToUpper())
```

Comment:
Rather than using `.ToUpper()`, you can use a case-insensitive hashset. e.g.:
`.ToImmutableHashSet(StringComparer.OrdinalIgnoreCase)`. That way you don't need to uppercase items before calling `.Contains()`